text/0000-clarity-api-description.md

```diff
@@ -0,0 +1,77 @@
+- Start Date: 2015-06-18
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+
+Provide clarity and additional elements/properites in API Description namespace.
+
+# Motivation
+
+The purpose of this document is to make it easier for newcomers to understand the format and build tools around it.
+
+# Detailed design
+
+## Combine Namespaces
+
+We should combine the API Description namespace with the resource namespace.
+Right now they are separate and it would be clearer if they were not.
+
+## Elements in Attributes
+
+Refract allows for including elements within the attributes section of an element.
+Even though it’s possible, it may be helpful to avoid this in such an important namespace.
+Reason being, this will be one of the first places people interact with Refract, and to use complex features may be too much.
+
+## HREF Variables
+
+We should call them “hrefVariables” everywhere instead of parameters in places (specifically transitions).
+I think this will be more consistent and easier to understand for newcomers.
+This will make more sense in how we handle inheritance as well, because parameters inheriting from hrefVariables does not seem optimal.
+
+## Defining Classes
+
+Swagger provides a way to define tags that can be applied to resources throughout the document.
+Refract has the “class” attribute that we can use the same way, we only need a way to define a class name and provide the description for it.
+Of course, this won’t carry over to API Blueprint, but that’s OK.
```

Comment:
@smizell my thinking here is that if we want to support both generic hierarchical grouping elements (e.g. categories in refract today, which is how we implement API Blueprint resource groups) **and** allow an item to be in multiple grouping elements (a resource that is a part of two groups aka tags like in Swagger) then a way to support both is by creating a generic hierarchical grouping element that uses references to define its members. Rather than having an element like a resource group which contains resources, we have a tag element that a resource can reference. This allows for really flexible resource organization and should support most formats. For example:

```apib
# Ungrouped Resource [/nogroup]

# Group Example Resources
Some description.

## My Resource [/foo]
## Another Resource [/bar]
```

Would become something like (even a bit simpler than I proposed above since we really don't need the `resources` or `tags` classes):

```json
["category", {"class": ["api"]}, {}, [
  ["resource", {"title": "Ungrouped Resource"}, {}, ...],
  ["resource", {"title": "My Resource", "tags": ["Example Resources"]}, {"href": "/foo"}, ...],
  ["resource", {"title": "Another Resource", "tags": ["Example Resources"]}, {"href": "/bar"}, ...],
  ["tag", {"title": "Example Resources", "description": "Some description."}, {}, null]
]]
```

Of course that assumes we use the tag title as its unique identifier. As a tooling author I can build a menu based on the tags and using Minim we can easily query for all elements tagged with `foo`:

* Ungrouped Resource
* Example Resources
  * My Resource
  * Another Resource

Does that make sense? What do you think? This is somewhat future-proof as it allows arbitrarily nested levels of groupings and resources for organization while keeping the refract representation simple(-ish). Any thoughts on this @zdne?